go/arrow/array/builder.go

```diff
@@ -277,6 +277,8 @@ func NewBuilder(mem memory.Allocator, dtype arrow.DataType) Builder {
 	case arrow.UNION:
 	case arrow.DICTIONARY:
 	case arrow.MAP:
+		typ := dtype.(*arrow.MapType)
```

Comment:
Looks like in this change, you're only adding map support, but here you are changing union and dictionary to use the map builder. I'm not sure in this context the difference between map and dictionary.

Reply:
you're still thinking in C++ context. Go doesn't automatically fallthrough like C++, this *only* modifies the Map case, it doesn't change anything about Union or Dictionary.

Reply:
ahh thanks